src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -38,7 +45,86 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(new DocumentDBAttributeBindingProvider(context.Config, this, context.Trace));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<DocumentDBAttribute, DocumentDBContext>(typeof(DocumentDBAsyncCollector<>),
```

Comment:
B/c we don't know the type of T at compile time. You can't specify DocumentDBAsyncCollector<T> (or use <>) as a generic parameter. So the method will use reflection to create this type for us at runtime.

But this is going away anyway -- it'll be my job to construct the collector myself via reflection. That gets rid of the need for us to care about your constructor parameters. That's in Mike's latest PR.

---
In reply to: [63895687](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/97#discussion_r63895687) [](ancestors = 63895687)